bin/API_Migration_scripts_readme/WebScraper_Java_Code.ipynb

```diff
@@ -0,0 +1,376 @@
+    "response = requests.get(url)\n",
+    "soup = BeautifulSoup(response.content, \"html.parser\")\n",
+    "\n",
+    "#getting links from the webpage\n",
+    "links = []\n",
+    "for a in soup.find_all(\"a\"):\n",
+    "    if a.has_attr(\"href\") and a[\"href\"].startswith(\"/sdk/api_diff/33/changes/\"):\n",
+    "        links.append(a[\"href\"])\n",
+    "\n",
+    "#creation and writing the file with DictWriter\n",
+    "csv_file = open(\"classes.csv\", \"w\", newline=\"\")\n",
+    "csv_writer = csv.DictWriter(csv_file, fieldnames=[\"Package Class\"])\n",
+    "csv_writer.writeheader()\n",
+    "\n",
+    "#technically not writing any links but the classes themselves\n",
+    "for link in links:\n",
+    "    css_class = link.split('/')[-1]\n",
+    "    csv_writer.writerow({\"Package Class\": css_class})\n",
+    "\n",
+    "\n",
+    "#JSON not required\n",
+    "json_file = open(\"classes.json\", \"w\")\n",
+    "json_data = {\"links\": links}\n",
+    "json.dump(json_data, json_file)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "355d9124",
+   "metadata": {},
+   "source": [
+    "## python script to compare the CSV against Java files in a given directory\n",
+    "##### python script to use the obtained CSV file. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2e458972",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def match_csv_java(csv_file, directory): #method accepting CSV and a directory\n",
+    "    java_files = os.listdir(directory)\n",
+    "    with open(csv_file, \"r\") as f: # open CSV in read mode\n",
+    "        reader = csv.reader(f)\n",
+    "        for row in reader:\n",
+    "            classes = row[1]         \n",
+    "    #java_files = os.listdir(directory)\n",
+    "            for java_file in java_files:\n",
+    "                if not os.path.isdir(java_file) and java_file.endswith(\".java\"): #to avoid error when encountering a directory - if another directory exists\n",
+    "                    with open(os.path.join(directory, java_file), \"r\", encoding=\"utf-8\") as f:\n",
+    "                        text = f.read()\n",
+    "                        if classes in text:\n",
+    "                            print(f\"Found {classes} in {java_file}\") # printing this in such a way we know which package/ class is in which file   \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c7b00cef",
+   "metadata": {},
+   "source": [
+    "### 1. data collection"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fba6313a",
+   "metadata": {},
+   "source": [
+    "#### primary folder"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "577a2047",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#absolute PATH: format - csv, directory                            \n",
+    "if __name__ == \"__main__\":\n",
+    "    match_csv_java(\"/tmp\",\"/tmp\")"
```

Comment:
again, why do we have all these copies of the same function with the same inputs?